presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayIndeterminateOperator.java

```diff
@@ -0,0 +1,122 @@
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.SqlType;
+import com.google.common.base.Throwables;
+import io.airlift.slice.Slice;
+
+import javax.annotation.Nullable;
+
+import java.lang.invoke.MethodHandle;
+
+import static com.facebook.presto.metadata.OperatorType.INDETERMINATE;
+
+@ScalarOperator(INDETERMINATE)
+public final class ArrayIndeterminateOperator
+{
+    private ArrayIndeterminateOperator() {}
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean indeterminate(
+            @OperatorDependency(operator = INDETERMINATE, returnType = StandardTypes.BOOLEAN, argumentTypes = {"T"}) MethodHandle function,
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") @Nullable Block block,
+            @IsNull boolean isNull)
+    {
+        if (isNull) {
+            return true;
+        }
+        for (int i = 0; i < block.getPositionCount(); i++) {
+            if (block.isNull(i)) {
+                return true;
+            }
+            try {
+                if ((boolean) function.invoke(type.getObject(block, i), false)) {
+                    return true;
+                }
+            }
+            catch (Throwable t) {
+                Throwables.propagateIfInstanceOf(t, Error.class);
+                Throwables.propagateIfInstanceOf(t, PrestoException.class);
+
+                throw new PrestoException(StandardErrorCode.INTERNAL_ERROR, t);
+            }
+        }
+        return false;
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = long.class)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean indeterminateLong(
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") @Nullable Block block,
+            @IsNull boolean isNull)
+    {
+        return indeterminateSpecialization(block, isNull);
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = boolean.class)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean indeterminateBoolean(
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") @Nullable Block block,
+            @IsNull boolean isNull)
+    {
+        return indeterminateSpecialization(block, isNull);
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = Slice.class)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean indeterminateSlice(
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") @Nullable Block block,
+            @IsNull boolean isNull)
+    {
+        return indeterminateSpecialization(block, isNull);
+    }
+
+    @TypeParameter("T")
+    @TypeParameterSpecialization(name = "T", nativeContainerType = double.class)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean indeterminateDouble(
```

Comment:
Have we determined whether these specializations matter? They all have the same implementation (i.e., they are not parameterized on a type & operator). 

The only thing that could virtualize is the call to `block.isNull()`. That would happen anyway, since a given type could see multiple block implementations.